peps/pep-0784.rst

```diff
@@ -0,0 +1,277 @@
+to CPython by Ma Lin, who also wrote the `output buffer implementation used in
+the standard library today <https://github.com/python/cpython/commit/f9bedb630e8a0b7d94e1c7e609b20dfaa2b22231>`_.
+The project has since been taken over by Rogdham and is published to PyPI. The
+APIs in ``pyzstd`` are similar to the APIs for other compression modules in the
+standard library such as ``bz2`` and ``lzma``.
+
+Minimum supported Zstandard version
+-----------------------------------
+
+The minimum supported Zstandard was chosen as v1.4.5, released in May of 2020.
+This version was chosen as a minimum based on reviewing the versions of
+Zstandard available in a number of Linux distribution package repositories,
+including LTS releases. This version choice is rather conservative to maximize
+compatibility with existing LTS Linux distributions, but a newer Zstandard
+version could likely be chosen given that newer Python releases are generally
+packaged as part of newer distribution releases.
+
+Specification
+=============
+
+The ``compression`` namespace
+-----------------------------
+
+A new namespace package for compression modules will be introduced named
+``compression``. The top-level module for this package will be empty to begin
+with, but a standard API for interacting with compression routines may be
+added in the future to the toplevel.
+
+The ``compression.zstd`` module
+-------------------------------
+
+A new module, ``compression.zstd`` will be introduced with Zstandard
+compression APIs that match other compression modules in the standard library,
+namely
+
+* ``compress`` / ``decompress`` - APIs for one-shot compression/decompression
+* ``ZstdFile`` / ``open`` - APIs for interacting with streams and file-like
+  objects
+* ``ZstdCompressor`` / ``ZstdDecompressor`` - APIs for incremental compression/
+  decompression
+
+It will also contain some Zstandard-specific functionality
+
+* ``ZstdDict`` / ``train_dict`` / ``finalize_dict`` - APIs for interacting with
+  Zstandard dictionaries, which are useful for compressing many small chunks of
+  similar data
+
+``libzstd`` optional dependency
+-------------------------------
+
+The ``libzstd`` library will become an optional dependency of CPython. If the
+library is not available, the ``compression.zstd`` module will be unavailable.
+This is handled automatically on Unix platforms as part of the normal build
+environment detection.
+
+On Windows, ``libzstd`` will be added to
+`the source dependencies <https://github.com/python/cpython-source-deps>`_
+used to build libraries CPython depends on for Windows.
+
+Other compression modules
+-------------------------
+
+New import names ``compression.lzma``, ``compression.bz2``, and
+``compression.zlib`` will be introduced re-exporting the contents of the
+existing ``lzma``, ``bz2``, and ``zlib`` modules respectively. Starting with
```

Comment:
Lets not emit a DeprecationWarning any time soon.  Plan that to happen several releases in the future.

lesson: hashlib put a DeprecationWarning in place for the md5 and sha modules on day 1.  it caused all sorts of noise and annoyance for people maintaining many-version compatible code.

I suggest turning the DeprecationWarning on after the last release of Python without the `compression.` module leaves maintenance.

Reply:
Thanks! This is really useful feedback and history/context. I like the idea of waiting until the last Python without `compression.` is dropped from support.